x/zoneconcierge/keeper/hooks.go

```diff
@@ -38,7 +38,12 @@ func (h Hooks) AfterRawCheckpointSealed(ctx context.Context, epoch uint64) error
 
 // AfterRawCheckpointFinalized is triggered upon an epoch has been finalised
 func (h Hooks) AfterRawCheckpointFinalized(ctx context.Context, epoch uint64) error {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	headersToBroadcast := h.k.getHeadersToBroadcast(ctx)
+	if len(headersToBroadcast) == 0 {
+		h.k.Logger(sdkCtx).Info("no new BTC headers to broadcast", "epoch", epoch)
+		return nil
+	}
```

Comment:
Why do we need this check?

In addition if `len(headersToBroadcast) == 0`, then the protocol will skip broadcasting BTC timestamps and update the last sent segment, which is not expected. Shall we remove it then?

Reply:
> Both broadcasting and updating last segment would get skipped - code link

I think we should not skip these two if `len(headersToBroadcast) == 0`. The protocol still needs to broadcast this new BTC timestamp. It's possible that there is no new BTC header since last BTC timestamp, but the chain receives a new BTC checkpoint that is `w` deep, e.g., vigilante reporter is offline for a long time. 

But indeed `lastSegment` is not supposed to be `nil` in this case -- the supposed behaviour is that the chain does not broadcast any BTC header, but in the current impl it will broadcast `w` [headers](https://github.com/babylonlabs-io/babylon/blob/d0133d28e1e45db7a2678c9fd3e868594d439447/x/zoneconcierge/keeper/ibc_packet_btc_timestamp.go#L168-L173). Maybe we need to fix this part